guides/drafts/incremental-adoption-guide.md

```diff
@@ -0,0 +1,115 @@
+
+## Step 2: Add `Store` service to your application
+
+You would need to create you own store service. Before it was automatically injected by Ember Data. 
+Here is how you do it:
+
+```ts
+// eslint-disable-next-line ember/use-ember-data-rfc-395-imports
+import Store from 'ember-data/store';
+import { service } from '@ember/service';
+import RequestManager from '@ember-data/request';
+
+export default class MyStore extends Store {
+  @service declare requestManager: RequestManager;
+}
+
+```
+
+Notice we still want to import the `Store` class from `ember-data/store` package. You might have a lint rule that says don't do it. You can disable it for this import. The reason we want to import it from `ember-data/store` is because we want to use Ember Data models, serializers, adapters, etc. and alongside we want to start utilizing new APIs.
+
+## Step 3: Add `RequestManager` service to your application
+
+Now let's create our very own `RequestManager` service. It is a service that is responsible for sending requests to the server. It is a composable class, which means you can add your own request handlers to it. In the example below we are adding `LegacyNetworkHandler`, `TestHandler` and `Fetch` handlers.
+
+```ts
+import RequestManager from '@ember-data/request';
+import Fetch from '@ember-data/request/fetch';
+import { LegacyNetworkHandler } from '@ember-data/legacy-compat';
+import { CacheHandler } from 'ember-data/store';
+import { setBuildURLConfig } from '@ember-data/request-utils';
+
+const TestHandler = {
+  async request({ request }, next) {
+    console.log('TestHandler.request', request);
+
+    const newContext = await next(request);
+
+    console.log('TestHandler.response after fetch', newContext.response);
+
+    return next(newContext);
+  },
+};
+
+export default class Requests extends RequestManager {
+  constructor(args) {
+    super(args);
+    this.use([LegacyNetworkHandler, TestHandler, Fetch]);
+    this.useCache(CacheHandler);
+  }
+}
+```
+
+Let's go over the code above:
+
+1. `LegacyNetworkHandler` is the handler that is responsible for sending requests to the server using the old APIs. It will interrupt handlers chain if it detects request using old APIs. It will process it as it used to be doing with Adapters/Fetch/Serializers workflow.
+
+2. Next is `TestHandler`. It is a handler that is responsible for logging requests. It is a quick example of how you can add your own handlers to the request manager. We will take a look at more useful examples later.
+
+3. Lastly `Fetch`. It is a handler that is responsible for sending requests to the server using the new Ember Data APIs. It must be last handler you put in the chain. After finishing request it will enrich handlers context with `response` property. And pass it back to the handlers chain in reverse order. So `TestHandler` will receive `response` property first, and so on if we would have any.
```

Comment:
```suggestion
3. Lastly `Fetch`. It is a handler that sends requests to the server using the `fetch` API. It expects responses to be JSON and when in use it should be the last handler you put in the chain. After finishing each request it will convert the response into json and pass it back to the handlers chain in reverse order as the request context's response. So `TestHandler` will receive `response` property first, and so on if we would have any.
```